stl/src/locale0.cpp

```diff
@@ -135,8 +135,36 @@ __PURE_APPDOMAIN_GLOBAL locale::id ctype<unsigned short>::id(0);
 
 __PURE_APPDOMAIN_GLOBAL locale::id codecvt<unsigned short, char, mbstate_t>::id(0);
 
+#define _Compiler_barrier() _STL_DISABLE_DEPRECATED_WARNING _ReadWriteBarrier() _STL_RESTORE_DEPRECATED_WARNING
+
+#if defined(_M_ARM) || defined(_M_ARM64) || defined(_M_ARM64EC)
+#define _Memory_barrier()             __dmb(0xB) // inner shared data memory barrier
+#define _Compiler_or_memory_barrier() _Memory_barrier()
+#elif defined(_M_IX86) || defined(_M_X64)
+// x86/x64 hardware only emits memory barriers inside _Interlocked intrinsics
+#define _Compiler_or_memory_barrier() _Compiler_barrier()
+#else // ^^^ x86/x64 / unsupported hardware vvv
+#error Unsupported hardware
+#endif // hardware
+
 _MRTIMP2_PURE const locale& __CLRCALL_PURE_OR_CDECL locale::classic() { // get reference to "C" locale
-    _Init();
+    const auto mem = reinterpret_cast<const intptr_t*>(&locale::_Locimp::_Clocptr);
+    intptr_t as_bytes;
+#ifdef _WIN64
+#ifdef _M_ARM
+    as_bytes = __ldrexd(_Mem);
+#else
+    as_bytes = __iso_volatile_load64(mem);
+#endif
+#else // ^^^ _WIN64 / !_WIN64 vvv
+    as_bytes = __iso_volatile_load32(mem);
+#endif // ^^^ !_WIN64 ^^^

```

Comment:
As previously discussed/feared, this file is indeed built for `/clr:pure` and the intrinsic usage fails to compile. I did a test build and for x86, the errors are:

```
D:\msvc\src\vctools\crt\github\stl\src\locale0.cpp(160,16): error C3861: '__iso_volatile_load32': identifier not found [D:\msvc\src\vctools\crt\managed\xmd\msvcurt_kernel32\msvcurt_kernel32.nativeproj]
D:\msvc\src\vctools\crt\github\stl\src\locale0.cpp(162,5): error C3861: '_ReadWriteBarrier': identifier not found [D:\msvc\src\vctools\crt\managed\xmd\msvcurt_kernel32\msvcurt_kernel32.nativeproj]
D:\msvc\src\vctools\crt\github\stl\src\locale0.cpp(193,9): error C3861: '_ReadWriteBarrier': identifier not found [D:\msvc\src\vctools\crt\managed\xmd\msvcurt_kernel32\msvcurt_kernel32.nativeproj]
D:\msvc\src\vctools\crt\github\stl\src\locale0.cpp(197,9): error C3861: '__iso_volatile_store32': identifier not found [D:\msvc\src\vctools\crt\managed\xmd\msvcurt_kernel32\msvcurt_kernel32.nativeproj]
```

(This is yet another reminder that we need to get around to porting the `/clr` and `/clr:pure` builds to GitHub - sorry about that.)

Can you guard these changes so that when `_M_CEE_PURE` is defined, nothing is changed? (There are virtually no users of `/clr:pure` so it doesn't need to receive any performance improvement, we just need to keep it compiling and working.)